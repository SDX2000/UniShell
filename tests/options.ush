#!/usr/bin/env unishell
echo "================"
echo $SCRIPT_NAME
echo "================"

options
echo
pushopt autoprint false
123
"Hello"
"World"
pushopt autoprint on
123
"Hello"
"World"
echo pushopt echo on
pushopt echo on
options
echo popopt echo
popopt echo
options
echo Auto print: $(peekopt autoprint)