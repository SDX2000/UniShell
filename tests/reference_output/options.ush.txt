================
options.ush
================
autoprint: ['True', 'False', 'True']
echo: ['False']
prompt: ['<lambda>']

False
123
Hello
World
True
123
Hello
World
pushopt echo on
True
autoprint: ['True', 'False', 'True']
echo: ['False']
prompt: ['<lambda>']
popopt echo
True
autoprint: ['True', 'False', 'True']
echo: ['False']
prompt: ['<lambda>']
Auto print: True
